roles/docker/tasks/main.yml

```diff
@@ -69,6 +69,22 @@
     reg_flag: --insecure-registry
   notify:
   - restart docker
+  
+- name: Set Proxy Settings
+  lineinfile:
+    dest: /etc/sysconfig/docker
+    regexp: '^{{ item.reg_conf_var }}=.*$'
+    line: "{{ item.reg_conf_var }}='{{ item.reg_fact_val }}'"
+  with_items:
+    - reg_conf_var: HTTP_PROXY
+      reg_fact_val: "{{ openshift.common.http_proxy | default('') }}"
+      when: "http_proxy in openshift.common"
+    - reg_conf_var: HTTPS_PROXY
+      reg_fact_val: "{{ openshift.common.https_proxy | default('') }}"
+    - reg_conf_var: NO_PROXY
+      reg_fact_val: "{{ openshift.common.no_proxy | default('') | join(',') }}"
```

Comment:
We shoudln't be referencing openshift facts in the docker role. Rather there should be docker_ variables for configuring the proxy and openshift_docker_facts should map the openshift fact values to the docker_ variables